src/Properties/ModelCastHelper.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace NunoMaduro\Larastan\Properties;
+
+use Illuminate\Contracts\Database\Eloquent\Castable;
+use Illuminate\Contracts\Database\Eloquent\CastsAttributes;
+use Illuminate\Contracts\Database\Eloquent\CastsInboundAttributes;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+use PHPStan\Reflection\ParameterReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Type\Accessory\AccessoryNumericStringType;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\BooleanType;
+use PHPStan\Type\Constant\ConstantIntegerType;
+use PHPStan\Type\FloatType;
+use PHPStan\Type\IntegerType;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+
+class ModelCastHelper
+{
+    public function __construct(protected ReflectionProvider $reflectionProvider)
+    {
+    }
+
+    public function getReadableType(string $cast, Type $originalType): Type
+    {
+        $cast = Str::before($cast, ':');
+
+        $attributeType = match ($cast) {
+            'int', 'integer' => new IntegerType(),
+            'real', 'float', 'double' => new FloatType(),
+            'decimal' => new AccessoryNumericStringType(),
+            'string' => new StringType(),
+            'bool', 'boolean' => new BooleanType(),
+            'object' => new ObjectType('stdClass'),
+            'array', 'json' => new ArrayType(new MixedType(), new MixedType()),
+            'collection' => new ObjectType('Illuminate\Support\Collection'),
+            'date', 'datetime' => $this->getDateType(),
+            'immutable_date', 'immutable_datetime' => new ObjectType('Carbon\CarbonImmutable'),
```

Comment:
Shouldn't we use `$this->getDateType()` here also? Or another new method.

Isn't it possible that it will return `DateTimeImmutable` instead of `CarbonImmutable`?

Reply:
I don't think this will ever return `DateTimeImmutable`, because [this line](https://github.com/laravel/framework/blob/9138ffdf886f130d7cf372ab165b650b201e28e2/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L781) will throw an error if `asDateTime` or `asDate` returns a `DateTime`, because it doesn't have the `toImmutable` method.

Reply:
I tested it on a project. Yeah it throws error exactly like you said. So this is fine 👍🏽